src/components/hooks/useElementPosition.js

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from 'react';
+
+/**
+ * Custom hook that receives an element reference and returns the position of the element
+ *
+ * @param {object} elementRef  Reference to the element
+ *
+ * @returns {DOMRect} Element position
+ */
+export default function useElementPosition(elementRef) {
+  const [position, setPosition] = useState(null);
+  const target = elementRef.current;
+
+  if (target && !position) {
```

Comment:
we want to run this code after each render, so we need to put it inside a useEffect
we also don't want to call setPosition if the position has not changed from the prev render:
```js
useEffect(() => {
  if (target) {
    const rect = target.getRect();
    if (!position || rect.top !== position.top || rect.left !== position.left || ....) {    
      setPosition(rect);
    }
}, [target]);

Reply:
useEffect() guarantees that the code inside is executed after the DOM has been updated, not part of the render function

> Sometimes, we want to run some additional code after React has updated the DOM.

https://reactjs.org/docs/hooks-effect.html#effects-without-cleanup